ncm-metaconfig/src/main/pan/components/metaconfig/schema.pan

```diff
@@ -86,6 +86,18 @@ type ${project.artifactId}_textrender_convert = {
 
 type caf_service_action = string with match(SELF, '^(restart|reload|stop_sleep_start)$');
 
+type ${project.artifactId}_actions = {
+    @{Always run, happens before possible modifications}
+    'pre' ? string
+    @{Always run, happens before possible modifications, the file content is passed om stdin.
```

Comment:
```suggestion
    @{Always run, happens before possible modifications, the new file content is passed on stdin.
```

Reply:
```suggestion
    @{Always run before possible modifications with the new (or unchanged) file content is passed on stdin. A failure will cancel any file modification, unless the command is prefixed with -.
```

Reply:
i find `new (or unchanged)` confusing. i would expect `new/changed (or unchanged)`,but maybe i need some sleep ;)
 
this is actually the generated content, and this runs before anything is written. on failure, it just stops, writing to disk or even comparing with previous content is not attempted (i think, i would have to look at the code in FileWriter).

Reply:
I was unsure about "unchanged", if it clouds the situation then let's just go for:

Always run before possible modifications with the generated file content is passed on stdin. A failure will cancel any file modification, unless the command is prefixed with -.